app/packages/core/src/components/Grid/Grid.tsx

```diff
@@ -1,191 +1,165 @@
-import Flashlight from "@fiftyone/flashlight";
-import { freeVideos } from "@fiftyone/looker";
+import styles from "./Grid.module.css";
+
+import type { Lookers } from "@fiftyone/state";
+
+import Spotlight, { type ID } from "@fiftyone/spotlight";
 import * as fos from "@fiftyone/state";
-import { stringifyObj, useDeferrer, useExpandSample } from "@fiftyone/state";
-import React, { useEffect, useLayoutEffect, useRef } from "react";
-import { useRecoilCallback, useRecoilValue } from "recoil";
+import React, {
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useRecoilValue } from "recoil";
 import { v4 as uuid } from "uuid";
-import useFlashlightPager from "../../useFlashlightPager";
-import { flashlightLooker } from "./Grid.module.css";
-import {
-  gridCropCallback,
-  pageParameters,
-  rowAspectRatioThreshold,
-} from "./recoil";
-import useResize from "./useResize";
-
-const Grid: React.FC<{}> = () => {
-  const [id] = React.useState(() => uuid());
-  const store = fos.useLookerStore();
-  const expandSample = useExpandSample(store);
-  const { init, deferred } = useDeferrer();
+import { gridCrop, gridSpacing, pageParameters } from "./recoil";
+import useAt from "./useAt";
+import useEscape from "./useEscape";
+import useRefreshers from "./useRefreshers";
+import useSelect from "./useSelect";
+import useSelectSample from "./useSelectSample";
+import useSpotlightPager from "./useSpotlightPager";
+import useThreshold from "./useThreshold";
+
+function Grid() {
+  const id = useMemo(() => uuid(), []);
+  const lookerStore = useMemo(() => new WeakMap<ID, Lookers>(), []);
```

Comment:
Can take it offline and flag for later, but I'm curious about the usage of `WeakMap` here with ID has the key...I feel like we aren't getting much out of it, and if anything, losing access to iterators, etc.

IIRC, weak maps/sets are useful because they don't add strong refs to the _keys_ (not values), and when all refs to the keys are lost, the _key_ is garbage collected along with the value (assuming no remaining strong refs to the value, which in this case is unlikely. I thought we'd be concerned about GC of looker instances, not IDs). 

```
const looker = new Looker();

const desc = {description};

lookerStore.add(desc, looker);
looker = null; // no remaining refs to looker except in the weakmap
// the following still yields looker instance
// because we still have a strong ref to looker in the weakmap (because it's a value).
// if this is the intended effect, then ignore my comment 😅 
lookerStore.get(desc); 
```

Reply:
So I will admit there are not tests that assert this setup. And it is an initial foray into this pattern...

The intention of the design is that only `@fiiftyone/spotlight` data structures, e.g. a `Row`, contain strong references to `ID` values. And the implication is rows are garbage collected when a `Spotlight` contracts, e.g. drops its backward section, mutates its forward section as the new backward section, and creates a fresh (empty) forward section.

So, in theory, the `ID`s are garbage collected through scrolling and `Looker` instances (and sample data in the other `WeakMap`) are garbage collected, as well

Though adding tests/better assurances around this is the next step 🚀